### ISSUE TITLE
[Bug]: Patreon RSS feeds dont work correctly

### ISSUE THREAD

**Issue by warwolf** [advplyr/audiobookshelf#2207]
### Describe the issue

So patreons RSS feeds for podcasts dont work correctly currently.
Every day it looses information about downloaded episodes.
Episodes are still downloaded in local files but not in detected as downloaded.

Possibly because RSS data contains items with dynamic urls that change between syncs?
This is the only relevant change in the RSS feed as i can see when comparing old xml-data with new.

```
<item>
	<title>499. No Such Thing As Old Testament Podcasts</title>
	<link>https://www.patreon.com/posts/499-no-such-as-90309652</link>
	<itunes:image href="DYNAMIC URL"/>
	<description>&lt;p&gt;Dan, James, Andrew and Lou Sanders discuss skating, shanties, flies and fishing.&lt;/p&gt;&lt;p&gt;Visit nosuchthingasafish.com for news about live shows, merchandise and more episodes.&lt;/p&gt;</description>
	<enclosure url="DYNAMIC URL" length="99411987" type="audio/mpeg"/>
	<guid isPermaLink="false">90309652</guid>
	<pubDate>Thu, 05 Oct 2023 23:01:00 GMT</pubDate>
</item>
```

### Steps to reproduce the issue

1. Find and download a few episodes of a podcast.
[Episodes downloaded](https://i.imgur.com/hbnQFVe.png)
2. A day later or when checking back no matching episodes are downloaded (in find episodes list that is)
[A day later](https://i.imgur.com/u4UwMgs.png)

### Audiobookshelf version

2.4.4

### How are you running audiobookshelf?

Docker

**Comment by advplyr** [comment]
We should be using the `GUID` but instead we are using the URL to match with existing episodes. This was brought up before here https://github.com/advplyr/audiobookshelf/issues/1896#issuecomment-1624650436 but I haven't looked into it yet

**Comment by advplyr** [comment]
Fixed in [v2.5.0](https://github.com/advplyr/audiobookshelf/releases/tag/v2.5.0)

Note that this is only fixed for episodes downloaded going forward. Existing episodes with this issue would need to be redownloaded to store the `GUID`.
